### DISCUSSION TITLE
Database not initializing properly

### DISCUSSION THREAD

**Discussion by AceIsHuman** [ujjwalguptaofficial/JsStore#296] [Q&A]
Creating a database with and without a web worker causes errors like `this.worker_.postMessage is not a function` and `workerRef.queryManager is not a constructor`.

Node v16.17.1
Electron v20.3.0
Vue 2.6.12

Jsstore did previously work on node v14.20.0. Could this be an issue related to Node, and is there any possible workaround while using v16.17.1?

**Comment by alisabrigok** [comment]
I'm having the same issue. 

For me, while working without a web worker via workerInjector it gives `workerRef.queryManager is not a constructor issue`. The full error print:

```
jsstore.commonjs2.js?b6ae:359 Uncaught TypeError: workerRef.QueryManager is not a constructor
    at ConnectionHelper.initQueryManager_ (jsstore.commonjs2.js?b6ae:359:1)
    at Object.setup (index.js?94b5:1:160)
    at Connection.addPlugin (jsstore.commonjs2.js?b6ae:820:1)
 ```

I can't upgrade the version post 4.3.10 due to this. 4.4.0 and above gives error.

I have a Webpack 4 configurated React app that was perfectly running with JsStore 4.3.9 (4.3.10 also works fine), then decided to upgrade to 4.4.4 but had `workerRef.queryManager is not a constructor` issue.

When I import "workerInjector" to use JsStore without web worker, I can see that Webpack is doing its job to import the module. Because when I console.log it, I see the printed workerInjector variable.

```
import workerInjector from 'jsstore/dist/worker_injector';

console.log(workerInjector) -> {setup: ƒ setup(r,e)}
```

I assume the problem occurred after the changes to the library related to compression or Webpack 5.

Any ideas? @ujjwalguptaofficial

**Comment by ujjwalguptaofficial** [comment]
yeah it must be broken after upgrading to webpack5. Will try to reproduce and fix ASAP.

**Comment by ujjwalguptaofficial** [comment]
@AceIsHuman @alisabrigok issue has been fixed in v - 4.4.5. Please install and let me know.

The issue got introduced due to webpack upgrade, i have also added some test cases - so that it don't ger reproduced again.

**Comment by alisabrigok** [comment]
I just installed version 4.4.5 and it fixed my issue. Thanks for the swift replies and fixes @ujjwalguptaofficial

**Comment by AceIsHuman** [comment]
Just updated to 4.4.5 and the issue seems to be fixed for `connection.initDb(database)` without web worker, using the workerInjector plugin.

I cannot get the database to initialize with a web worker. The following code does not produce an error, it just gets hung up on line 5 at `initDb`.

```
// connection.js
1 import { Connection } from 'jsstore'
2 import * as JsStoreWorker from 'jsstore/dist/jsstore.worker.commonjs2.js'
3
4 const connection = new Connection(new Worker(JsStoreWorker.default))
5 const isDbCreated = await connection.initDb(getDatabase())
```

I have also tried creating a preload script in place of the import on line 2. This produces the same result.
```
// preload.js
const workerPath = require('file-loader?name=scripts/[name].[hash].js!jsstore/dist/jsstore.worker.js').default

// connection.js
4 const connection = new Connection(new Worker(workerPath))
```

**Reply by ujjwalguptaofficial** [reply]
@AceIsHuman can you please give me something reproducable. I am not able to get from your snippets where things can be wrong.

**Reply by AceIsHuman** [reply]
I created a repo with the error here. https://github.com/AceIsHuman/jsstore-webworker-issue

**Reply by ujjwalguptaofficial** [reply]
the worker is not being initialized properly @AceIsHuman , i checked out the repo  and i see html code inside jsstore worker.

![image](https://user-images.githubusercontent.com/10938615/196477859-6c46687e-22d1-4975-ab24-74cbacf654ae.png)

**Reply by AceIsHuman** [reply]
Any idea why this might happen? When I go to the actual worker path that is in the console log, the js file does contain the worker script. 
![image](https://user-images.githubusercontent.com/47653241/196479150-ed64c000-81b9-40fd-a5b5-323e9590927f.png)

**Reply by AceIsHuman** [reply]
I get the same result when using:
```
import * as JsStoreWorker from 'jsstore/dist/jsstore.worker.commonjs2.js'
const jsStoreWorker = new Worker(JsStoreWorker.default)
```

![image](https://user-images.githubusercontent.com/47653241/196487857-a68765bf-3c21-4e76-80b6-677f50afd720.png)
